pallets/common/src/protocol_fee.rs

```diff
@@ -35,14 +36,22 @@ pub enum ProtocolOp {
     IdentityAddSecondaryKeysWithAuthorization,
     PipsPropose,
     VotingAddBallot,
+    ContractsPutCode,
 }
 
 /// Common interface to protocol fees for runtime modules.
-pub trait ChargeProtocolFee<AccountId> {
+pub trait ChargeProtocolFee<AccountId, Balance> {
     /// Computes the fee of the operation and charges it to the given signatory.
     fn charge_fee(op: ProtocolOp) -> DispatchResult;
 
     /// Computes the fee for `count` similar operations, and charges that fee to the given
     /// signatory.
     fn batch_charge_fee(op: ProtocolOp, count: usize) -> DispatchResult;
+
+    /// Used for charging the instantiation fee for the smart extension.
+    fn charge_extension_instantiation_fee(
+        fee: Balance,
+        owner: AccountId,
+        network_share: Perbill,
+    ) -> DispatchResult;
```

Comment:
I think this should remain outside protocol_fee since it's quite different. Protocol fees are decided and managed by the governance committee while the instantiation fee are controlled by the contract author.

Reply:
It is only used to provides essential utilities to the fee deduction. the committee doesn't have any authority to change the instantiation_fee and network _share, Data always lives in the wrapper contract pallet